giraph-core/src/test/java/org/apache/giraph/comm/RequestTest.java

```diff
@@ -67,251 +71,280 @@
-  /** Configuration */
-  private ImmutableClassesGiraphConfiguration conf;
-  /** Server data */
-  private ServerData<IntWritable, IntWritable, IntWritable> serverData;
-  /** Server */
-  private NettyServer server;
-  /** Client */
-  private NettyClient client;
-  /** Worker info */
-  private WorkerInfo workerInfo;
-
-  @Before
-  public void setUp() {
-    // Setup the conf
-    GiraphConfiguration tmpConf = new GiraphConfiguration();
-    GiraphConstants.COMPUTATION_CLASS.set(tmpConf, IntNoOpComputation.class);
-    conf = new ImmutableClassesGiraphConfiguration(tmpConf);
-
-    @SuppressWarnings("rawtypes")
-    Context context = mock(Context.class);
-    when(context.getConfiguration()).thenReturn(conf);
-
-    // Start the service
-    serverData = MockUtils.createNewServerData(conf, context);
-    serverData.prepareSuperstep();
-    workerInfo = new WorkerInfo();
-    server = new NettyServer(conf,
+  abstract public static class SharedSetup {
+    /** * Configuration */
+    protected ImmutableClassesGiraphConfiguration conf;
+    /** * Server data */
+    protected ServerData<IntWritable, IntWritable, IntWritable> serverData;
+    /** * Server */
+    protected NettyServer server;
+    /** * Client */
+    protected NettyClient client;
+    /** * Worker info */
+    protected WorkerInfo workerInfo;
+
+    @Before
+    public void setUp() {
+      // Setup the conf
+      GiraphConfiguration tmpConf = new GiraphConfiguration();
+      GiraphConstants.COMPUTATION_CLASS.set(tmpConf, IntNoOpComputation.class);
+      conf = new ImmutableClassesGiraphConfiguration(tmpConf);
+
+      @SuppressWarnings("rawtypes")
+      Context context = mock(Context.class);
+      when(context.getConfiguration()).thenReturn(conf);
+
+      // Start the service
+      serverData = MockUtils.createNewServerData(conf, context);
+      serverData.prepareSuperstep();
+      workerInfo = new WorkerInfo();
+      server = new NettyServer(conf,
         new WorkerRequestServerHandler.Factory(serverData), workerInfo,
-            context, new MockExceptionHandler());
-    server.start();
+        context, new MockExceptionHandler());
+      server.start();
 
-    workerInfo.setInetSocketAddress(server.getMyAddress(), server.getLocalHostOrIp());
-    client = new NettyClient(context, conf, new WorkerInfo(),
+      workerInfo.setInetSocketAddress(server.getMyAddress(), server.getLocalHostOrIp());
+      client = new NettyClient(context, conf, new WorkerInfo(),
         new MockExceptionHandler());
-    server.setFlowControl(client.getFlowControl());
-    client.connectAllAddresses(
+      server.setFlowControl(client.getFlowControl());
+      client.connectAllAddresses(
         Lists.<WorkerInfo>newArrayList(workerInfo));
+    }
   }
 
-  @Test
-  public void sendVertexPartition() {
-    // Data to send
-    int partitionId = 13;
-    Partition<IntWritable, IntWritable, IntWritable> partition =
+  public static class NonParameterizedTest extends SharedSetup {
```

Comment:
Any reason for not using Parameterized tests for this two test cases?

Reply:
These other tests are a bit different with respect to what the parameter would be and the range of values. For now, I just wanted to make this particular test parameterized.